clients/da-vinci-client/src/main/java/com/linkedin/davinci/consumer/VeniceChangelogConsumerImpl.java

```diff
@@ -371,13 +382,18 @@ protected Collection<PubSubMessage<K, ChangeEvent<V>, VeniceChangeCoordinate>> i
       long timeoutInMs,
       String topicSuffix) {
     List<PubSubMessage<K, ChangeEvent<V>, VeniceChangeCoordinate>> pubSubMessages = new ArrayList<>();
+    List<Integer> partitionsToFilter = new ArrayList<>();
     ConsumerRecords<KafkaKey, KafkaMessageEnvelope> consumerRecords = kafkaConsumer.poll(timeoutInMs);
```

Comment:
I think  we might need to add lock here, poll is the most-frequent consumer call in this class.

Reply:
What race condition are we concerned about with poll?

Reply:
Just thinking aloud about it, the interactions we'd be worried about would be poll getting called before or after partition assignments.  Partitions assignment in versionswap is done in line with the poll call, so we can put that aside (assuming the underlying kafka client won't do something funny with concurrent poll calls on the same client?).  So the next is seek/subscribe by the user.  If poll is called before an assignment, thats ok,because the next poll call will get the data, and if a poll is called after an assignment, thats just what we'd expect.  I don't think we need to accomodate for these two calls being very close together?

That is unless theres is some bad behavior with double polls calls that I'm not thinking of?

Reply:
I think we probably will not go into double polls, but one thing we need to be careful is the `consumer#assign` and `consumer#poll`, as kafka consumer is not thread safe, and `consumer#assign` is in a future api call, this might need the future api call is blocked.